base-math/truncdddf.c

```diff
@@ -66,15 +66,25 @@ CONVERT_WRAPPER(
 	          return SIGNBIT(a) ? -INFINITY : INFINITY;
 	      }
 	  }
-	else if (exp < -BINPOWOF10_LIMIT)	/* Obvious underflow.  */
+	else if (exp <= -BINPOWOF10_DENORM_DBL)	/* Obvious underflow.  */
 	  {
 	    if (DFP_EXCEPTIONS_ENABLED)
 	      DFP_HANDLE_EXCEPTIONS (FE_UNDERFLOW|FE_INEXACT);
 
 	    switch (fegetround())
 	      {
 	        case FE_TONEAREST:
-	          return SIGNBIT(a) ? -__DBL_DENORM_MIN__ : __DBL_DENORM_MIN__;
+	          mant = llabs (a_norm);
+	          if (exp < -BINPOWOF10_DENORM_DBL ||
+	                    ((exp == -BINPOWOF10_DENORM_DBL) &&
+	                     (mant >= 24703282292062300)))
+	           {
+	              return SIGNBIT(a) ? -0.0 : 0.0;
+	           }
+	          else
+	           {
+	              return SIGNBIT(a) ? -__DBL_DENORM_MIN__ : __DBL_DENORM_MIN__;
+	           }
```

Comment:
Likewise.  Please review if the the line won't pass the limit of 79 characters.

Reply:
It's 79? I was think that was 80.

Reply:
(n < 80) or (n <= 79).
Source: https://www.gnu.org/prep/standards/html_node/Formatting.html

Reply:
But do we follow gnu standards in this project?
It wasn't my understanding since https://github.com/libdfp/libdfp/pull/59#discussion_r135768144

Reply:
@tuliom Ok it will be fixed.
@gftg85 I guess there's not really any standard in this code. There's lots of weird stuffs also lots of trailing whitespace. Fix that should be a task of this own.

Reply:
@gftg85 When libdfp was written, it did not follow the GNU coding standard.
There are cases where 8 spaces are replaced with tabs, and cases that are not.
While we don't have a clear standard, I think we have to target the GNU one (for glibc compatibility) limited by current coding standard already in use in a file (without having different coding styles in the same file).